File: 1126_MSFT_Reactor_AI102_S16/src/TS/palindromeChecker.ts
// palindromeChecker.ts

export function isPalindromeTS(s: string): boolean {
    if (s === null || s === "") {
        return false;
    }

    let left = 0;
    let right = s.length - 1;

    while (left < right) {
        console.log(`Current elements left index: '${s[left]}' and right index: '${s[right]}'`);

        if (s[left] !== s[right]) {
            console.log("The elements aren't the same, hence we return false");
            return false;
        }

        left++;
        right--;
    }

    return true;
}

File: 1126_MSFT_Reactor_AI102_S16/src/Python/palindrome_checker.py
# palindrome_checker.py

def is_palindrome(s):
    if s is None or s == "":
        return False

    left, right = 0, len(s) - 1

    while left < right:
        print(f"Current elements left index: '{
              s[left]}' and right index: '{s[right]}'")

        if s[left] != s[right]:
            print("The elements aren't the same, hence we return False")
            return False

        left += 1
        right -= 1

    return True
